### DISCUSSION TITLE
Add About button to installer's welcome screen and Pre-extract the file to be open with the button

### DISCUSSION THREAD

**Discussion by CadMan77** [wixtoolset/issues#7577] [Questions]
I want to customize the welcome screen of installer for my product to add "About" button that would open some descriptive document (ex. About-<Product Name>.pdf).

In particular I want to know two things:
1. Is it even possible to customize welcome screen (add PushButton)?
2. How to get/extract the PDF file copy from the MSI to [TempFolder] before display the customized welcome screen?

PS:
I'm using WiX Toolset 3.11

PPS:
Is there more actual and/or detailed book (I don't like to learn software by videos) than Wix 3.6 Developer's Guide ... by Nick Ramirez (of 2012)?

**Comment by CadMan77** [comment]
OK. I believe the answer to my question#1 is "NO - customizing standard Welcome Dialog with WiX is impossible." (the answer is based on this article - https://wixtoolset.org/docs/v3/wixui/wixui_customizations/).
The customization desired is not listed.
Any ideas on how to achieve my goal? Switch from WiX to some alternative MSI-building software? Which one?

**Reply by robmen** [reply]
It's absolutely possible. You can go as far as to write your own custom wizard sequence. The standard UIs provided by WiX are written in WiX code. You can see them in the source code.

**Reply by CadMan77** [reply]
@robmen, thank for your inspiring comment - I've tried a bit harder and finally succeeded with adding the button (#1 is almost done).
Would you, please, be so kind as to help me a bit with my question #2. Should I pre-unpack the pdf with some Custom Action? Can you share link to some relevant sample (if any). Thank you in advance.

**Reply by robmen** [reply]
Meh. Your second scenario seems strange and I didn't have any interest in spending my free time to understand it.

**Reply by robmen** [reply]
To be more specific why I'm not interested, I wouldn't try to do what you are doing with the PDF. Use RTF. That's supported by the Windows Installer. Then you are done.

**Reply by CadMan77** [reply]
Thank you for this clarification. I've chosen PDF as format due to it's 'native write protection' (I believe it should be clear what I mean). I don't mind to switch to RTF but I still don't understand where to start from ...

**Reply by BMurri** [reply]
For an example, the license dialog hosts RTF.

**Reply by CadMan77** [reply]
I've already customized the LicenseAgreement dialog in my installer (use my own EULA.RTF).
I tried to replace reference to my EULA.RTF with reference to my About.RTF and installer does show it*. But I don't want to show About doc this way (on each installation).
I want to provide user with option to call it by extra pushbutton of Welcome screen.

*
Apart from raster images I've used in my About.RTF (it shows filenames on image placeholders).

**Reply by BMurri** [reply]
What I was thinking was you create a new dialog based on the current license dialog, except it hosts About.RTF. The new button pops up this new dialog. That takes care of the write protection you were concerned about and eliminates the need for a new custom action.

**Comment by barnson** [comment]
Take a look at https://wixtoolset.org/docs/tools/wixext/util/#launch-an-embedded-file.

**Reply by CadMan77** [reply]
@barnson 
Wow!
Using WixShellExecBinary seems the most suitable way to achieve my goal.
But how do I get WixToolset.Util.wixext (which my SetupProject is supposed to reference)?
I'm using Visual Studio Community 2022 (17.6.4) with "WiX v3 - Visual Studio 2022 Extension" (all my SetupProjects till now were created with WiX Toolset v3).
WixShellExecBinary was introduced in WiX Toolset v4, right?

Since  WiX Toolset v4 got released I decided it is worth to try it.
I've installed "HeatWave for VS2022 Extension" (yet I'm not sure it is the WiX Toolset v4 itself).

I also tried to search extensions for "WixToolset.Util.wixext" but get 0 results...

... I'm a bit puzzled.

UPDATE:
I believe I've found the WixToolset.Util.wixext Repository (https://www.nuget.org/packages/WixToolset.Util.wixext/).
I then tried to install the package using different approaches but none was successful:
![Untitled](https://github.com/wixtoolset/issues/assets/47462051/798a53af-a96b-4c23-9280-0a37fd7961b9)
What I'm missing?

**Reply by barnson** [reply]
Works for me using the NuGet Package Manager in VS2022.

**Reply by BMurri** [reply]
`WixShellExec` exists in both v3 and v4. `WixShellExecBinary` is new to v4. There's no interest in backporting anything to v3, without cash changing hands.

**Reply by CadMan77** [reply]
@BMurri 
Are you sure that 'WixShellExecBinary' is new to v.4?
I have just found 'WixShellExecBinary' mentioned with relation to WiX Toolset v.3.6**(https://github.com/AnalogJ/Wix3.6Toolset/blob/master/RC0-source/wix36-sources/src/ext/UtilExtension/wixlib/UtilExtension.wxs)
Yet I still don't understand whether (and how) I can use it for project created with v.3.11...

**
All new is a well-forgotten old ?

**Reply by BMurri** [reply]
I was referring to the documentation. Here are the v3 docs for the ShellExec custom action

https://wixtoolset.org/docs/v3/customactions/shellexec/
https://wixtoolset.org/docs/v3/howtos/ui_and_localization/run_program_after_install/
